src/uri.rs

```diff
@@ -389,6 +389,48 @@ impl StdError for UriError {
     }
 }
 
+// TODO: Is ByteStr adding value? Inline it?
```

Comment:
I find it adds value, because it encapsulates the safety into a separate type, so that `Uri` doesn't really need to think about safety.